src/com/goide/inspections/GoStructInitializationInspection.java

```diff
@@ -22,94 +22,148 @@
+    return o == null || Comparing.equal(o, objectToCompare);
+  }
+
+  @NotNull
+  private static List<String> getFieldDefinitionsNames(@NotNull GoStructType type) {
+    return type.getFieldDeclarationList().stream()
+      .flatMap(declaration -> getFieldDefinitionsNames(declaration).stream())
+      .collect(toList());
+  }
+
+  @NotNull
+  private static List<String> getFieldDefinitionsNames(@NotNull GoFieldDeclaration declaration) {
+    GoAnonymousFieldDefinition definition = declaration.getAnonymousFieldDefinition();
+    return definition != null ? list(definition.getName()) : map(declaration.getFieldDefinitionList(), GoNamedElement::getName);
+  }
+
+  @Override
+  public JComponent createOptionsPanel() {
+    return new SingleCheckboxOptionsPanel("Report for local type definitions as well", this, "reportLocalStructs");
   }
 
   private static class GoReplaceWithNamedStructFieldQuickFix extends LocalQuickFixBase {
-    private String myStructField;
 
-    public GoReplaceWithNamedStructFieldQuickFix(@NotNull String structField) {
+    public GoReplaceWithNamedStructFieldQuickFix() {
       super(REPLACE_WITH_NAMED_STRUCT_FIELD_FIX_NAME);
-      myStructField = structField;
     }
 
     @Override
     public void applyFix(@NotNull Project project, @NotNull ProblemDescriptor descriptor) {
-      PsiElement startElement = descriptor.getStartElement();
-      if (startElement instanceof GoElement) {
-        startElement.replace(GoElementFactory.createLiteralValueElement(project, myStructField, startElement.getText()));
-      }
+      PsiElement element = ObjectUtils.tryCast(descriptor.getStartElement(), GoElement.class);
+      GoLiteralValue literal = element != null && element.isValid() ? PsiTreeUtil.getParentOfType(element, GoLiteralValue.class) : null;
+      GoStructType structType = getStructType(literal);
+      List<GoElement> elements = structType != null ? literal.getElementList() : emptyList();
+      List<String> fieldDefinitionNames = structType != null ? getFieldDefinitionsNames(structType) : emptyList();
+      if (!areElementsNamesMatchesDefinitions(getNames(elements), fieldDefinitionNames)) return;
+      replaceElementsByNamed(project, elements, fieldDefinitionNames);
+    }
+  }
+
+  private static void replaceElementsByNamed(@NotNull Project project,
+                                             @NotNull List<GoElement> elements,
+                                             @NotNull List<String> fieldDefinitionNames) {
+    for (int i = 0; i < elements.size(); i++) {
+      GoElement element = elements.get(i);
+      String fieldDefinitionName = GoPsiImplUtil.getByIndex(fieldDefinitionNames, i);
+      GoValue value = fieldDefinitionName != null && element.getKey() == null ? element.getValue() : null;
+      if (value == null) continue;
+
+      GoElement namedElement = GoElementFactory.createLiteralValueElement(project, fieldDefinitionName, value.getText());
+      element.replace(namedElement);
```

Comment:
Yes, indeed. But I could not find a way to add colon and key to the element without several boilerplate lines of code so I would like to stay with the current solution.